### DISCUSSION TITLE
Cannot instantiate interface Lcobucci\JWT\Decoder

### DISCUSSION THREAD

**Discussion by robertolanza93** [lcobucci/jwt#920] [Q&A]
Hello,
I'm working with magento 2.4.4 and Lcobucci 4.2.1
when we try to use Lcobucci\JWT\Token\Parser we receive the following error:

Cannot instantiate interface Lcobucci\JWT\Decoder in /var/www/magento2/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php

can someone please help with this problem?

Thanks,
Roberto

**Comment by Ocramius** [comment]
Duplicates https://github.com/lcobucci/jwt/discussions/744
Duplicates https://github.com/lcobucci/jwt/discussions/754

**Reply by robertolanza93** [reply]
Thank you for your reply!

Well, actually I have seen those problems:
- as for the 744 we are using the **Lcobucci\JWT\Token** one 
- as for the 754 the problem is not on the "Key" interface, but it is on Decoder interface that it is used on Lcobucci\JWT\Token\Parser class

did I miss something?

**Reply by lcobucci** [reply]
@robertolanza93 how is magento aware of the instances? Is that exception coming from a dependency injection system? If so, have you configured the correct implementation for the decoder interface?

**Reply by robertolanza93** [reply]
we have added this in the uses
 **use Lcobucci\JWT\Token\Parser**
and then we've added the Parser class into the constructor
doing this returns us the problem written above, without it, our class is working correctly....

**Reply by lcobucci** [reply]
Alright, that explains it. Magento is trying to automatically instantiate things on your behalf (because you added the constructor parameter) and it's unable to resolve the dependencies (because it's clueless about them).

You need to learn how that works in Magento to be able to properly configure it. Alternatively, you may use the high-level implementation and not bother about Magento configuration.

More info: https://lcobucci-jwt.readthedocs.io/en/stable/quick-start/#parsing-tokens
